sandbox/pc2sandbox.sh

```diff
@@ -0,0 +1,234 @@
+fi
+
+# set the specified memory limit - input is in MB, cgroup v2 requires bytes, so multiply by 1M
+# but only if > 0.
+# "max" means unlimited, which is the cgroup v2 default
+DEBUG echo checking memory limit
+if [ "$MEMLIMIT" -gt "0" ] ; then
+  DEBUG echo setting memory limit to $MEMLIMIT MB
+  echo $(( $MEMLIMIT * 1024 * 1024 ))  > $PC2_SANDBOX_CGROUP_PATH/memory.max
+  echo 1  > $PC2_SANDBOX_CGROUP_PATH/memory.swap.max
+else
+  DEBUG echo setting memory limit to max, meaning no limit
+  echo "max" > $PC_SANDBOX_CGROUP_PATH/memory.max  
+  echo "max" > $PC_SANDBOX_CGROUP_PATH/memory.swap.max  
+fi
+
+# set the specified CPU time limit - input is in secs, cgroup v2 requires usec, so multiply by 1M.
+# cgroup v2 expects two parameters:  absolute time and "period", but if only one is provided it is "absolute time"
+DEBUG echo setting cpu limit to $TIMELIMIT seconds
+ulimit -t $TIMELIMIT
+TIMELIMIT_US=$((TIMELIMIT * 1000000))
+#echo $TIMELIMIT_US  > $PC2_SANDBOX_RUN_CGROUP_PATH/cpu.max
+
+DEBUG echo setting maximum user processes to 32 + whatever the user is currently using
+ulimit -u $((32+`ps -T -u $USER | wc -l`))
+
+#put the current process (and implicitly its children) into the pc2sandbox cgroup.
+DEBUG echo putting $$ into $PC2_SANDBOX_CGROUP_PATH cgroup
+if ! echo $$ > $PC2_SANDBOX_CGROUP_PATH/cgroup.procs
+then
+	echo Failed. quitting.
+	exit 1
+fi
+
+# run the command
+# the following are the cgroup-tools V1 commands; need to find cgroup-tools v2 commands
+# echo Using cgexec to run $COMMAND $*
+# cgexec -g cpu,memory:/pc2 $COMMAND $*
+
+# since we don't know how to use cgroup-tools to execute, just execute it directly (it's a child so it
+#  should still fall under the cgroup limits).
+DEBUG echo Executing $COMMAND $* 
+
+# Set up trap handler to catch wall-clock time exceeded and getting killed by PC2's execute timer
+trap HandleTerminateFromPC2 15
+
+$COMMAND $* <&0 &
+# Remember child's PID for possible killing off later
+submissionpid=$!
+# Wait for child
+wait $submissionpid
+
+COMMAND_EXIT_CODE=$?
+
+# See if we were killed due to memory - this is a kill 9 if it happened
+
+kills=`grep oom_kill $PC2_SANDBOX_CGROUP_PATH/memory.events | cut -d ' ' -f 2`
+
+if test "$kills" != "0"
+then
+	DEBUG echo The command was killed due to out of memory
+	COMMAND_EXIT_CODE=${FAIL_MEMORY_LIMIT_EXCEEDED}
+else
+	# Get cpu time
+	cputime=`grep usage_usec $PC2_SANDBOX_CGROUP_PATH/cpu.stat | cut -d ' ' -f 2`
+	if test "$cputime" -gt "$TIMELIMIT_US"
+	then
+		DEBUG echo The command was killed due to out of CPU Time "($cputime > $TIMELIMIT_US)"
+		COMMAND_EXIT_CODE=${FAIL_TIME_LIMIT_EXCEEDED}
+	elif test "$COMMAND_EXIT_CODE" -ge 128
+	then
+		DEBUG echo The command terminated abnormally with exit code $COMMAND_EXIT_CODE
+	fi
+fi
+DEBUG echo Finished executing $COMMAND $*
+DEBUG echo $COMMAND exited with exit code $COMMAND_EXIT_CODE
+DEBUG echo
+
+# TODO: determine how to pass pc2sandbox.sh results back to PC2...
```

Comment:
I'm not sure, but it appears that the above line of code is left over from the original version of the script (which I wrote).  If the decision is "we're going to pass the sandbox results back to PC2 in the form of the command exit code (as done by the following lines), the above comment should be removed.

Reply:
I think the intent of the comment (and I left it there on purpose) was to give a more detailed explanation of what happened.  LIke, how much memory was used?  what the times were in case of timeout.  (basically, the stuff in the debug log...)

Reply:
Clarified the comment in the script.